pkg/languages/defaults/gradle.go

```diff
@@ -0,0 +1,112 @@
+package defaults
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/draft/pkg/reporeader"
+	log "github.com/sirupsen/logrus"
+)
+
+const SOURCE_COMPATIBILITY = "sourceCompatibility"
+const TARGET_COMPATIBILITY = "targetCompatibility"
+const SERVER_PORT = "server.port"
+const GRADLE_FILE_FORMAT = "*.gradle"
+
+type GradleExtractor struct {
+}
+
+// GetName implements reporeader.VariableExtractor
+func (*GradleExtractor) GetName() string {
+	return "gradle"
+}
+
+// MatchesLanguage implements reporeader.VariableExtractor
+func (*GradleExtractor) MatchesLanguage(lowerlang string) bool {
+	return lowerlang == "gradle" || lowerlang == "gradlew"
+}
+
+// ReadDefaults implements reporeader.VariableExtractor
+func (*GradleExtractor) ReadDefaults(r reporeader.RepoReader) (map[string]string, error) {
+	separatorsSet := createSeparatorsSet()
+	cutSet := createCutSet()
+	extractedValues := make(map[string]string)
+	files, err := r.FindFiles(".", []string{GRADLE_FILE_FORMAT}, 2)
+	if err != nil {
+		return nil, fmt.Errorf("error finding gradle files: %v", err)
+	}
+	if len(files) > 0 {
+		f, err := r.ReadFile(files[0])
+		if err != nil {
+			log.Warn("Unable to read gradle file, skipping detection")
+			return nil, nil
+		}
+		content := string(f)
+		// this separator is used to split the line from build.gradle ex: sourceCompatibility = '1.8'
+		// output will be ['sourceCompatibility', '1.8'] or ["sourceCompatibility", "1.8"]
+		separatorFunc := func(c rune) bool {
+			return separatorsSet.Contains(c)
+		}
+		// this func takes care of removing the single or double quotes from split array output
+		cutset := func(c rune) bool { return cutSet.Contains(c) }
+		if strings.Contains(content, SOURCE_COMPATIBILITY) || strings.Contains(content, TARGET_COMPATIBILITY) || strings.Contains(content, SERVER_PORT) {
+			stringAfterSplit := strings.FieldsFunc(content, separatorFunc)
+			for i, s := range stringAfterSplit {
+				if s == SOURCE_COMPATIBILITY && i+1 < len(stringAfterSplit) {
```

Comment:
since every potential action checks `i<len(stringAfterSplit)` could that just go before the ifs with a break like : `if i+1>=len(stringAfterSplit) { break }`?
then we can simplify the logic checks in the if/else chain

Reply:
@davidgamero I can do that but still the if/else remains since it needs to check for source_comtability or target_compatbility or server.port. is that okay ?

Reply:
done.